server/socket/index.ts

```diff
@@ -1,147 +1,104 @@
-          }
-        } else {
-          log.error('received malformed message');
-        }
-      } catch (e) {
-        log.error('received malformed message');
-      }
-    });
-
-    ws.on(WebsocketEventType.Close, () => {
-      log.trace(`Received close event for websocket ${sessionId}`);
-      if (sessionMap[sessionId]) {
-        delete sessionMap[sessionId];
-      }
-      if (userId && userMap[userId]) {
-        delete userMap[userId];
-      }
-    });
-  });
-
-  // TODO: maybe unify these, or else remove the event type from payload and add it manually here?
-  wss.on(WebsocketMessageType.InitializeScrollback, (payload: IWebsocketsPayload<any>, userIds: number[]) => {
-    if (logging) log.info(`Payloading ${JSON.stringify(payload)} to users ${JSON.stringify(userIds)}`);
-    for (const user of userIds) {
-      if (user && user in userMap && userMap[user].isAlive) {
-        userMap[user].send(JSON.stringify(payload));
-      }
-    }
-  });
-
-  wss.on(
-    WebsocketMessageType.Notification,
-    (payload: IWebsocketsPayload<any>, notifications: { [user: number]: any }) => {
-      if (logging) {
-        log.info(`Payloading ${JSON.stringify(payload)} to users ${JSON.stringify(Object.keys(notifications))}`);
-      }
-      for (const [ user, notification ] of Object.entries(notifications)) {
-        if (user && user in userMap && userMap[user].isAlive) {
-          // augment notification with unique database id
-          payload.data.id = notification.id;
-          payload.data.subscription_id = notification.subscription_id;
-          userMap[user].send(JSON.stringify(payload));
-        }
-      }
-    }
+export function setupWebSocketServer(httpServer: http.Server) {
+	// since the websocket servers are not linked with the main Commonwealth server we do not send the socket.io client
+// library to the user since we already import it + disable http long-polling to avoid sticky session issues
+	const io = new Server(httpServer, { transports: ['websocket'], cors: {
+			origin,
+			methods: ["GET", "POST"]
+		}});
+
+	io.use(authenticate);
+
+	io.on('connection', (socket) => {
+		log.trace(`${socket.id} connected`)
+		socket.on('disconnect', () => {
+			log.trace(`${socket.id} disconnected`);
+		});
+	});
+
+	io.engine.on('connection_error', (err) => {
+		// log.error(err.req);      // the request object
+		// console.log(err.code);     // the error code, for example 1
+		// console.log(err.message);  // the error message, for example "Session ID unknown"
+		// console.log(err.context);  // some additional error context
+		log.error("A WebSocket connection error has occurred", err)
+	})
+
+// create the chain-events namespace
+	const ceNamespace = createCeNamespace(io);
+
+	// enables the admin analytics dashboard (creates /admin namespace)
+	instrument(io, {
+		auth: false,
+	})
+
+	const pool = new Pool({
+		connectionString: DATABASE_URI,
+		ssl: {
+			rejectUnauthorized: false,
+		},
+	})
+	pool.query(`
```

Comment:
Can you explain what this database code is doing?

Reply:
Creates a connection pool (similar to sequelize). The connection pool is used by Socket.io to ensure consistency between different Socket.io servers. In other words, this allows Socket.io to use our Postgres database as relaying mechanism between the Socket.io servers.

Reply:
Oh I see, neat!